src/main/java/org/gridsuite/securityanalysis/server/repositories/CriteriaUtils.java

```diff
@@ -77,10 +91,24 @@ private static Predicate filterToAtomicPredicate(CriteriaBuilder criteriaBuilder
                 case CONTAINS -> criteriaBuilder.like(criteriaBuilder.upper(stringExpression), "%" + escapedFilterValue.toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
                 case STARTS_WITH -> criteriaBuilder.like(criteriaBuilder.upper(stringExpression), escapedFilterValue.toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
                 case EQUALS -> criteriaBuilder.equal(criteriaBuilder.upper(stringExpression), stringValue.toUpperCase());
+                default ->
+                        throw new UnsupportedOperationException("This type of filter is not supported for text data type");
+            };
+        }
+
+        if (ResourceFilterDTO.DataType.NUMBER == filter.dataType()) {
+            Double valueDouble = Double.valueOf((String) value);
+            return switch (filter.type()) {
+                case NOT_EQUAL -> criteriaBuilder.notEqual(getColumnPath(path, field), valueDouble);
+                case LESS_THAN_OR_EQUAL -> criteriaBuilder.lessThanOrEqualTo(getColumnPath(path, field), valueDouble);
+                case GREATER_THAN_OR_EQUAL ->
+                        criteriaBuilder.greaterThanOrEqualTo(getColumnPath(path, field), valueDouble);
+                default ->
+                        throw new UnsupportedOperationException("This type of filter is not supported for number data type");
             };
```

Comment:
```suggestion
        if (ResourceFilterDTO.DataType.NUMBER == filter.dataType()) {
            Double valueDouble = Double.valueOf((String) value);
            return switch (filter.type()) {
                case NOT_EQUAL -> criteriaBuilder.notEqual(expression, valueDouble);
                case LESS_THAN_OR_EQUAL -> criteriaBuilder.lessThanOrEqualTo(expression, valueDouble);
                case GREATER_THAN_OR_EQUAL ->
                        criteriaBuilder.greaterThanOrEqualTo(expression, valueDouble);
                default ->
                        throw new UnsupportedOperationException("This type of filter is not supported for number data type");
            };
```

expression is already the calculated Path to the field, we don't need to add "path" and "field" to params